pulp_2to3_migration/app/plugin/docker/migrator.py

```diff
@@ -137,17 +137,35 @@ class InterrelateContent(Stage):
         """
         Relate each item in the input queue to objects specified on the DeclarativeContent.
         """
-        async for dc in self.items():
-
-            if dc.extra_data.get('man_rel'):
-                self.relate_manifest_to_list(dc)
-            elif dc.extra_data.get('blob_rel'):
-                self.relate_blob(dc)
-
-            if dc.extra_data.get('config_blob_rel'):
-                self.relate_config_blob(dc)
-
-            await self.put(dc)
+        async for batch in self.batches():
+            manifestlist_manifest_batch = []
+            blob_manifest_batch = []
+            manifest_batch = []
+            with transaction.atomic():
+                for dc in batch:
+                    if dc.extra_data.get('man_rel'):
+                        thru = self.relate_manifest_to_list(dc)
+                        manifestlist_manifest_batch.extend(thru)
+                    elif dc.extra_data.get('blob_rel'):
+                        thru = self.relate_blob(dc)
+                        blob_manifest_batch.extend(thru)
+
+                    if dc.extra_data.get('config_blob_rel'):
+                        manifest_to_update = self.relate_config_blob(dc)
+                        manifest_batch.append(manifest_to_update)
```

Comment:
It seems there is always only one manifest to update, isn't it?
I understand the unification, I'm just not sure if it's worth creating a list and then call a bulk_create if we know there will be only one item. It's also confusing, because the code makes it read that there  can be multiple relations to save.

Let me know if I miss anything, I'm not confident in my container plugin knowledge :)

Reply:
that's correct, manifest_to_update contains only one item, however we are working here with batches were in one batch we will process multiple items of manifests, meaning that manifest_batch will have X of manifests